python/cudf/cudf/utils/utils.py

```diff
@@ -341,6 +341,17 @@ def is_na_like(obj):
     return obj is None or obj is cudf.NA or obj is cudf.NaT
 
 
+def _is_null_host_scalar(slr) -> bool:
+    if is_na_like(slr):
+        return True
+    elif (
+        isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr)
+    ) or slr is pd.NaT:
+        return True
+    else:
+        return False
```

Comment:
Try to combine this into something like:
```python
return is_na_like(slr) or (isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr)
```

Reply:
Or if you leave it the long way, maybe add comments that help explain the second case.